OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,475 @@
+            patcher.AddPrefix<DoubleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisRemovePre));
+            patcher.AddPostfix<SingleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisUpdatePost));
+            patcher.AddPostfix<DoubleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisUpdatePost));
+            Instance = this;
+        }
+
+        private void UpdateCombo()
+        {
+            if (Time.realtimeSinceStartup - _lastUpdate > _updateCooldown)
+            {
+                _lastUpdate = Time.realtimeSinceStartup;
+                Int64 hash = 0;
+                int[] keys = new int[_maxComboLength];
+                int keysCount = 0;
+                bool countdownTrigger = false;
+                for (int code = _minUsefulKey; code < _maxUsefulKey; code++)
+                {
+                    if (Enum.IsDefined(typeof(KeyCode), (KeyCode)code) && UnityEngine.Input.GetKey((KeyCode)code))
+                    {
+                        keys[keysCount] = code;
+                        keysCount++;
+                        if (keysCount > _maxComboLength)
+                        {
+                            hash = (int)RegistrationCode.CombinationTooLong;
+                            break;
+                        }
+                        hash = hash * _maxUsefulKey + code;
+                        if (Time.realtimeSinceStartup - _timeout[code] < _cooldown)
+                            countdownTrigger = true;
+                    }
+                }
+                if (_comboRegistry.ContainsKey(hash))
+                {
+                    IModInputCombination combination = _comboRegistry[hash];
+                    if (combination == _lastPressed || !countdownTrigger)
+                    {
+                        if (_lastPressed != null)
+                            _lastPressed.SetPressed(false);
+                        _lastPressed = (ModInputCombination)combination;
+                        _lastPressed.SetPressed();
+                        for (int i = 0; i < keysCount; i++)
+                            _timeout[keys[i]] = Time.realtimeSinceStartup;
+                        return;
+                    }
+                }
+                if (_lastPressed != null)
+                {
+                    _lastPressed.SetPressed(false);
+                }
+                _lastPressed = null;
+            }
+        }
+
+        private bool IsPressed_Combo(IModInputCombination combination)
+        {
+            UpdateCombo();
+            return _lastPressed == combination;
+        }
+
+        private bool IsNewlyPressed_Combo(IModInputCombination combination, bool keep = false)
+        {
+            UpdateCombo();
+            return _lastPressed == combination && combination.IsFirst(keep);
+        }
+
+        private bool WasTapped_Combo(IModInputCombination combination)
+        {
```

Comment:
These are private but not used...?

Reply:
see above, I asked if should make make _Combo versions public

Reply:
Can's find the comment, can you link to it? Click "...", then "Copy link"

Reply:
https://github.com/amazingalek/owml/pull/94#issuecomment-621971295

Reply:
basically _Combo works like that: it gets an array of currently pressed buttons, generates hash and checks if any of registered key combinations match it exactly. 
There's is also _Single version that takes InputCombination's list of single-key "combinations" and checks if any of those is currently pressed and doesn't need to be ignored. This version was added to make it possible to activate multiple single-key "combinations" simultaneously
Default versions work by combining both

Reply:
single-key "combination" sounds weird though

Reply:
Idea behind make _Combo public is that some moddera might want for certain things to activate only when pressed keys match combination exactly. 
Probably would replace _Combo with _Strict or _Exact to make it more descriptive

Reply:
@amazingalek

Reply:
Since they're private and unused now, does that mean you could just remove them?
If they must be used from outside, I guess it's best to just make them public.

Reply:
Also, doesn't the underscore in the method names go against the style in the rest of the OWML codebase? Or does it serve a purpose I'm not aware of?

Reply:
removed underscore. I made Exact (previously _Combo) private because I thought it might confuse others
So what do you think: go for flexibility and make it public (probably with added XML documentation to reduce confusion) or keep it private and delete unused?

Reply:
If deleting them because they're not used is an option then do that :P 
Otherwise: public, no underscore 🤷

Reply:
decided to just make them public